### PR TITLE
Introduce Interface for Course Preference Choices on Profile Page

### PR DESCRIPTION
**Pull request by benbeauchamp7**
Includes enhancements for the profile page, most notably the ability to select preferred sections. It also features a small change to fetching API data and some information about what labs you are scheduled for.

This PR also includes the ability to save and load a schedule from the server.

### CODE REVIEW

File: src/Components/Profile/PreferenceBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  if (linkIDs !== null && linkIDs.length === 0) classes += " alert";
+  if (interacted) {
+    classes += " interacted " + edge;
+    classes += (bottom)? ' bottom' : '';
+  }
+  else classes += " grow-h";
+
+  return (
+    <div ref={ref} className={classes} onClick={() => setInteracted(true)}
+      title={`${course_instance.course_number}-${course_instance.section_number}`}
```
Comment (dantebarbieri) [inline src/Components/Profile/PreferenceBlock.tsx:121]: Can we also display the professor's name here? It's stored in the DB as `placeholder_professor_name` and should likely be returned by an API somewhere.

Comment (dantebarbieri) [reply src/Components/Profile/PreferenceBlock.tsx:121]: The idea for this, by the way, is that the professor teaching a course is often a motivator. We added that column to the DB for this purpose in particular.

File: src/Components/Profile/PreferenceBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  return (
+    <div ref={ref} className={classes} onClick={() => setInteracted(true)}
+      title={`${course_instance.course_number}-${course_instance.section_number}`} 
+      style={{...color, ...isVisible}}
+    >
+      { interacted? 
+        <div className="pref-pane">
+          <div className="pref-pane-title">Select courses</div>
+          <div key={`pref-row-checkall`} className="pref-row">
+            <div className="check-all" onClick={() => checkAll(course_instance.section_id)}>Select All</div>
```
Comment (dantebarbieri) [inline src/Components/Profile/PreferenceBlock.tsx:128]: Can we style this differently to look like a button with MUI-like or Bootstrap-like styling? It is underlined, but my eyes glance over it as text.

File: src/Components/Profile/ProfileSidebar.tsx [hunk 3]
```diff
@@ -1,11 +1,83 @@
...
+          </div>
+          <div>
+            {formatDate(block.start_time)}-{formatDate(block.end_time)} {block.days.join('')}
+          </div>
+          <div>
+            {block.place}
+          </div>
+        </div>
+      )
+    })
```
Comment (dantebarbieri) [inline src/Components/Profile/ProfileSidebar.tsx]: Does this fill the sidebar below the name with the assigned courses?
image
It seems empty, so if this doesn't do that, can we consider doing that? In general, how can I see what this is doing?

**Review by dantebarbieri — CHANGES_REQUESTED** [review]
Looks like there's good progress here! I requested a few changes.

File: src/Components/Profile/PreferenceBlock.tsx (hunk 2 above)
Comment (benbeauchamp7) [reply src/Components/Profile/PreferenceBlock.tsx:128]: MUI is a no, I'd need some ideas on how to style it. I was thinking a big obvious button would make the pane look stupid but I'm not sure how to make a slim clickable thing other than by underlining it

File: src/Components/Profile/ProfileSidebar.tsx (hunk 3 above)
Comment (benbeauchamp7) [reply src/Components/Profile/ProfileSidebar.tsx]: Try generating a schedule and checking back to that page

**Review by dantebarbieri — APPROVED** [review]
Overall, looks good to me and works well on my local

**Pull request merged**